python/core/auto_generated/qgsdataitem.sip.in

```diff
@@ -214,15 +221,19 @@ method.
 .. deprecated:: since QGIS 3.6. This method is no longer used by QGIS and will be removed in QGIS 4.0.
 %End
 
-    virtual bool rename( const QString &name );
+ virtual bool rename( const QString &name ) /Deprecated/;
```

Comment:
The implementations of rename and delete methods like to display message boxes asking for confirmation / showing information about successful operation. I can imagine we could have gui-free implementations of these which don't have user interaction and just do their job - in that case it would make sense to un-deprecate them...